paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/FlinkJobRecoveryITCase.java

```diff
@@ -0,0 +1,320 @@
+                triggerCheckpointAndWaitForWrites(jobClient, beforeRecoverExpectedRows.size());
+        jobClient.cancel().get();
+
+        List<Row> rows = batchSql(beforeRecoverCheckSql);
+        assertThat(rows.size()).isEqualTo(beforeRecoverExpectedRows.size());
+        assertThat(rows).containsExactlyInAnyOrder(beforeRecoverExpectedRows.toArray(new Row[0]));
+
+        for (String sql : updateSql) {
+            batchSql(sql);
+        }
+
+        Configuration config =
+                sEnv.getConfig()
+                        .getConfiguration()
+                        .set(StateRecoveryOptions.SAVEPOINT_PATH, checkpointPath);
+        for (Map.Entry<String, String> entry : recoverOptions.entrySet()) {
+            config.setString(entry.getKey(), entry.getValue());
+        }
+
+        jobClient = sEnv.executeSql(afterRecoverSql).getJobClient().get();
+        triggerCheckpointAndWaitForWrites(jobClient, afterRecoverExpectedRows.size());
+        jobClient.cancel().get();
+
+        rows = batchSql(afterRecoverCheckSql);
+        assertThat(rows.size()).isEqualTo(afterRecoverExpectedRows.size());
+        assertThat(rows).containsExactlyInAnyOrder(afterRecoverExpectedRows.toArray(new Row[0]));
+    }
+
+    private void createTargetTable(String tableName, BucketMode bucketMode) {
+        switch (bucketMode) {
+            case HASH_FIXED:
+                batchSql(
+                        String.format(
+                                "CREATE TABLE IF NOT EXISTS `%s` (k INT, f1 STRING, pt STRING, PRIMARY KEY(k, pt) NOT ENFORCED) WITH ('bucket'='2', 'commit.force-create-snapshot'='true')",
+                                tableName));
+                return;
+            case HASH_DYNAMIC:
+                batchSql(
+                        String.format(
+                                "CREATE TABLE IF NOT EXISTS `%s` (k INT, f1 STRING, pt STRING, PRIMARY KEY(k, pt) NOT ENFORCED) WITH ('bucket'='-1', 'commit.force-create-snapshot'='true')",
+                                tableName));
+                return;
+            case CROSS_PARTITION:
+                batchSql(
+                        String.format(
+                                "CREATE TABLE IF NOT EXISTS `%s` (k INT, f1 STRING, pt STRING, PRIMARY KEY(k) NOT ENFORCED) WITH ('bucket'='-1', 'commit.force-create-snapshot'='true')",
+                                tableName));
+                return;
+            case BUCKET_UNAWARE:
+                batchSql(
+                        String.format(
+                                "CREATE TABLE IF NOT EXISTS `%s` (k INT, f1 STRING, pt STRING) WITH ('bucket'='-1', 'commit.force-create-snapshot'='true')",
+                                tableName));
+                return;
+            default:
+                throw new IllegalArgumentException("Unsupported bucket mode: " + bucketMode);
+        }
+    }
+
+    private Snapshot waitForNewSnapshot(String tableName, long initialSnapshot)
```

Comment:
Actual value of parameter 'tableName' is always 'target_table' 

Private static final String xxx = "target_table";

Reply:
My mistake. I should have used `target_table2` in the second check in `testRestoreFromSavepointWithIgnoreSinkState` . Resolved.